docs/pages/advanced_usage.md

```diff
@@ -0,0 +1,80 @@
+# Advanced usage
+
+This repository is designed both for performance and for having a low memory footprint.
+Therefore, it provides bulk operations and the possibility to access objects as streams.
+We **strongly suggest** to use these methods if you use the `disk-objecstore` as a library,
+unless you are absolutely sure that objects always fit in memory, and you never have to
+access tens of thousands of objects or more.
+
+## Bulk access
+
+We continue from the commands of the basic usage. We can get the content of more objects at once:
+
+```python
+container.get_objects_content([hash1, hash2])
+# Output: {'6a96df63699b6fdc947177979dfd37a099c705bc509a715060dbfd3b7b605dbe': b'some_content',  'cfb487fe419250aa790bf7189962581651305fc8c42d6c16b72384f96299199d': b'some_other_content'}
+```
+
+For many objects (especially if they are packed), retrieving in bulk can give orders-of-magnitude speed-up.
+
+## Using streams
+
+### Interface
+
+First, let's look at the interface:
+
+```python
+with container.get_object_stream(hash1) as stream:
+    print(stream.read())
+# Output: b'some_content'
+```
+
+For bulk access, the syntax is a bit more convoluted (the reason is efficiency, as discussed below):
+
+```python
+with container.get_objects_stream_and_meta([hash3, hash1, hash2]) as triplets:
+    for hashkey, stream, meta in triplets:
+        print("Meta for hashkey {}: {}".format(hashkey, meta))
+        print("  Content: {}".format(stream.read()))
+```
+
+whose output is:
+
+```
+Meta for hashkey 6a96df63699b6fdc947177979dfd37a099c705bc509a715060dbfd3b7b605dbe: {'type': 'packed', 'size': 12, 'pack_id': 0, 'pack_compressed': False, 'pack_offset': 0, 'pack_length': 12}
+  Content: b'some_content'
+Meta for hashkey cfb487fe419250aa790bf7189962581651305fc8c42d6c16b72384f96299199d: {'type': 'packed', 'size': 18, 'pack_id': 0, 'pack_compressed': False, 'pack_offset': 12, 'pack_length': 18}
+  Content: b'some_other_content'
+Meta for hashkey d1e4103ce093e26c63ce25366a9a131d60d3555073b8424d3322accefc36bf08: {'type': 'loose', 'size': 13, 'pack_id': None, 'pack_compressed': None, 'pack_offset': None, 'pack_length': None}
+  Content: b'third_content'
+```
+
+**IMPORTANT NOTE**: As you see above, the order of the triplets **IS NOT** the order in which you passed the hash keys to
+`get_objects_stream_and_meta`. The reason is efficiency: the library will try to keep a (pack) file open as long as possible, and read it in order, to exploit efficiently disk caches.
```

Comment:
````suggestion
```{important}
As you see above, the order of the triplets **IS NOT** the order in which you passed the hash keys to
`get_objects_stream_and_meta`. The reason is efficiency: the library will try to keep a (pack) file open as long as possible, and read it in order, to exploit efficiently disk caches.
```
````

don't know if you realised, but you can use directive's like this which are equivalent to those in RST: https://myst-parser.readthedocs.io/en/latest/syntax/syntax.html#directives-a-block-level-extension-point

Reply:
Thanks! Now that you say, I remember having seen this in the AiiDA docs - this is the first time I use myst so I'm slowly learning! Thanks.